File: backend/assessment-service/src/main/java/com/sandbox/assessment/assignment/service/AssignmentService.java
package com.sandbox.assessment.assignment.service;

import com.sandbox.assessment.assignment.dto.AssignmentRequestDto;
import com.sandbox.assessment.assignment.dto.AssignmentResponseDto;
import com.sandbox.assessment.assignment.enums.AssignmentStatus;

import org.springframework.data.domain.Page;

import java.util.List;

/**
 * ==========================================================
 * Member 4
 * Assignment Service Interface
 * ==========================================================
 */
public interface AssignmentService {

    /**
     * Assign assessment to candidate.
     */
    AssignmentResponseDto assignAssessment(AssignmentRequestDto requestDto);

    /**
     * Get all assignments.
     */
    List<AssignmentResponseDto> getAllAssignments();

    /**
     * Get assignment by ID.
     */
    AssignmentResponseDto getAssignmentById(Long assignmentId);

    /**
     * Start assigned assessment.
     */
    AssignmentResponseDto startAssessment(Long assignmentId);

    /**
     * Submit assessment.
     */
    AssignmentResponseDto submitAssessment(Long assignmentId);

    /**
     * Delete assignment.
     */
    void deleteAssignment(Long assignmentId);
    
    /**
     * Returns all assignments of one candidate.
     */
    List<AssignmentResponseDto> getAssignmentsByCandidate(Long candidateId);

    /**
     * Returns all candidates assigned to one assessment.
     */
    List<AssignmentResponseDto> getAssignmentsByAssessment(Long assessmentId);
    
    Page<AssignmentResponseDto> getAssignments(
            int page,
            int size,
            String sortBy
    );
    
    /**
     * Get assignments by status.
     */
    List<AssignmentResponseDto> getAssignmentsByStatus(
            AssignmentStatus status);

    /**
     * Get assignments by candidate and status.
     */
    List<AssignmentResponseDto> getAssignmentsByCandidateAndStatus(
            Long candidateId,
            AssignmentStatus status);
}

File: frontend/src/services/AssignmentService.js
import AssessmentApi from "../api/AssessmentApi";

const AssignmentService = {

    assignAssessment(data) {
        return AssessmentApi.post("/assignments", data);
    },

    getAllAssignments() {
        return AssessmentApi.get("/assignments");
    },

    getAssignmentById(id) {
        return AssessmentApi.get(`/assignments/${id}`);
    },

    getAssignmentsByCandidate(candidateId) {
        return AssessmentApi.get(`/assignments/candidate/${candidateId}`);
    },

    getAssignmentsByStatus(status) {
        return AssessmentApi.get(`/assignments/status/${status}`);
    },

    getDashboardAnalytics() {
        return AssessmentApi.get("/assignment/dashboard");
    },

    getStatistics() {
        return AssessmentApi.get("/assignments/statistics");
    },

    startAssessment(id) {
        return AssessmentApi.put(`/assignments/${id}/start`);
    },

    submitAssessment(id) {
        return AssessmentApi.put(`/assignments/${id}/submit`);
    },

    deleteAssignment(id) {
        return AssessmentApi.delete(`/assignments/${id}`);
    }
};

export default AssignmentService;